Use ArrayRef<Scalar> for List of Scalar in codegen

wenleix opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #53583 Pass Scalar by reference
* #53582 Treat Scalar parameter as if it is constant
* **#53581 Use ArrayRef<Scalar> for List of Scalar in codegen**

Summary:
We will pass `Scalar` by reference in the following commit.

For a list of Scalar, `ScalarList` will be used as the type,
otherwise `ArrayRef<const Scalar&>` will be generated as the type
by default.

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D26904443](https://our.internmc.facebook.com/intern/diff/D26904443)

wenleix committed:
Introduce ScalarList as ArrayRef<Scalar>

Summary:
We will pass `Scalar` by reference in the following commit.

For a list of Scalar, `ScalarList` will be used as the type,
otherwise `ArrayRef<const Scalar&>` will be generated as the type
by default.

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

Commit:
Update on "Introduce ScalarList as ArrayRef<Scalar>"


Summary:
We will pass `Scalar` by reference in the following commit.

For a list of Scalar, `ScalarList` will be used as the type,
otherwise `ArrayRef<const Scalar&>` will be generated as the type
by default.

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D26904443](https://our.internmc.facebook.com/intern/diff/D26904443)

[ghstack-poisoned]

Commit:
Update on "Introduce ScalarList as ArrayRef<Scalar>"


Summary:
We will pass `Scalar` by reference in the following commit.

For a list of Scalar, `ScalarList` will be used as the type,
otherwise `ArrayRef<const Scalar&>` will be generated as the type
by default.

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D26904443](https://our.internmc.facebook.com/intern/diff/D26904443)

[ghstack-poisoned]

smessmer commented:
as discussed offline, let's just use `ArrayRef<Scalar>` directly instead of introducing `ScalarList`.

smessmer reviewed (changes requested):
-

Commit:
Update on "Introduce ScalarList as ArrayRef<Scalar>"


Summary:
We will pass `Scalar` by reference in the following commit.

For a list of Scalar, `ScalarList` will be used as the type,
otherwise `ArrayRef<const Scalar&>` will be generated as the type
by default.

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D26904443](https://our.internmc.facebook.com/intern/diff/D26904443)

[ghstack-poisoned]

wenleix commented:
Squashed into https://github.com/pytorch/pytorch/pull/53583 as it's now just one line change.
